fix: resolve config extensions using only one resolver (#1527)
Redocly/redocly-cli: .changeset/many-steaks-hope.md

---
"@redocly/openapi-core": patch
---

> Changed the `loadConfig` function to avoid overwriting the `externalRefResolver` internally. That improves caching external config resources.

Reviewer: Could you re-visit this explanation? What changed for the user?

Author: @lornajane Nothing really changes from the user's prospective. This is more of an internal change which helps improving performance in other tools (like the VS Code extension, Realm &c.).

```suggestion
Fixed the use of `externalRefResolver` in the `loadConfig` function, which improves caching external config resources.
```

Reviewer: I don't think the user has much knowledge of our internal naming, so how about 
```suggestion
Improved caching for external configuration resources.
```

Author: Much better, thanks!
